### DISCUSSION TITLE
solver tutorial

### DISCUSSION THREAD

**Discussion by carlo-hue** [phoebe-project/phoebe2#881] [Questions]
Good morning,

I have a very basic question....I'm running the tutorial http://phoebe-project.org/docs/2.4/tutorials/solver but I can't completely reproduce the graph immediately following the instruction:

_ = b.plot(solution='my_lcgeom_solution', show=True)

The graph is shown but the vertical lines, both dotted and solid, do not appear.
I also get the error:

ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.

could you help me?

thanks

**Comment by kecnry** [comment]
I suspect this is a compatibility issue with a newer version of either matplotlib or numpy - can you please let us know what version of matplotlib and numpy you have installed?

```
import matplotlib, numpy
print(f"{matplotlib.__version__=} {numpy.__version__=}")
```

Can you also send more of the traceback (if there is more information displayed showing where that `ValueError` is coming from)?

**Comment by carlo-hue** [comment]
Here versions:

matplotlib.__version__='3.7.2' numpy.__version__='1.24.3'

more of traceback:

 File ~\anaconda3\envs\my_phoebe\lib\site-packages\spyder_kernels\py3compat.py:356 in compat_exec
    exec(code, globals, locals)

  File c:\users\carlo\onedrive\carlo personale\documents\carlo\astronomia\binarie\programmi\tutorial solver.py:38
    _ = b.plot(solution='my_lcgeom_solution', show=True)

  File ~\anaconda3\envs\my_phoebe\lib\site-packages\phoebe\parameters\parameters.py:5955 in plot
    return self._show_or_save(save, show, animate,

  File ~\anaconda3\envs\my_phoebe\lib\site-packages\phoebe\parameters\parameters.py:6073 in _show_or_save
    fig = afig.draw(i=time,

  File ~\anaconda3\envs\my_phoebe\lib\site-packages\phoebe\dependencies\autofig\figure.py:606 in draw
    axesi.draw(ax=ax, i=i, calls=calls,

  File ~\anaconda3\envs\my_phoebe\lib\site-packages\phoebe\dependencies\autofig\axes.py:1250 in draw
    artists = call.draw(ax=ax, i=i,

  File ~\anaconda3\envs\my_phoebe\lib\site-packages\phoebe\dependencies\autofig\call.py:1097 in draw
    data = np.array([x, y])

ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.

**Reply by kecnry** [reply]
ok, thanks.  I added the details to a note in #636 which is tracking all these compatibility issues.  For now, would it be possible to downgrade matplotlib to 3.5.1 and/or numpy to 1.23.1 and let us know if that avoids the issue for you?

**Reply by carlo-hue** [reply]
ok, thanks, I'll try the other versione tomorrow and I'll write here the results

**Comment by carlo-hue** [comment]
Hi,

I downgraded matplotlib and numpy: it does work !

a question: with previous versions, the error I get only concerns the display and not the correctness of the results obtained, correct?

thanks a lot

**Reply by kecnry** [reply]
great, thanks for the update and hopefully we can get the compatibility issue fixed sometime soon.  Yes, this should only be an issue with plotting itself - if you see any changes in the _values_ themselves, please do let us know as that would point to a much more critical bug.
